packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -2549,8 +2549,9 @@ export class MergeTree {
         client: Client,
     ): ReferencePosition {
         if (isRemoved(segment)) {
-            if (!refTypeIncludesFlag(refType, ReferenceType.SlideOnRemove)) {
-                throw new UsageError("Can only create SlideOnRemove local reference position on a removed segment");
+            if (!refTypeIncludesFlag(refType, ReferenceType.SlideOnRemove | ReferenceType.Transient)) {
```

Comment:
the existing undo redo implementation uses transient references, which are always safe, as they not bi-directionally linked, they are only valid at the point in time they are created.